File: src/order-system/internal/models/order.go
package models

import (
	"fmt"
	"time"
)

type Order struct {
	OrderID    int64     `json:"order_id"`
	UserID     int64     `json:"user_id"`
	Category   string    `json:"category"`
	Amount     int       `json:"amount"`
	UnitPrice  float64   `json:"unit_price"`
	TotalPrice float64   `json:"total_price"`
	CreatedAt  time.Time `json:"created_at"`
}

func (o *Order) CalculateTotal() {
	o.TotalPrice = float64(o.Amount) * o.UnitPrice
}

type OrderPlacedCommand struct {
	MessageID   string    `json:"message_id"`
	MessageType string    `json:"message_type"`
	Order       Order     `json:"order"`
	CreatedAt   time.Time `json:"created_at"`
}

func NewOrderPlacedCommand(order Order) *OrderPlacedCommand {
	return &OrderPlacedCommand{
		MessageID:   generateID(),
		MessageType: "OrderPlacedCommand",
		Order:       order,
		CreatedAt:   time.Now(),
	}
}

func generateID() string {
	return fmt.Sprintf("%d", time.Now().UnixNano())
}

File: src/OrderSystem/OrderSystem.DomainModel/Models/Order.cs
namespace OrderSystem.OrderGen.Models;

public class Order
{
    public long OrderId { get; set; }
    public long UserId { get; set; }
    public string Category { get; set; } = string.Empty;
    public int Amount { get; set; }
    public decimal UnitPrice { get; set; }
    public decimal TotalPrice { get; set; }
    public DateTime CreatedAt { get; set; }

    public void CalculateTotal()
    {
        TotalPrice = Amount * UnitPrice;
    }
}